wooting-macro-backend/src/lib.rs

```diff
@@ -281,15 +293,20 @@ async fn execute_macro(macros: Macro, channel: UnboundedSender<rdev::EventType>)
 /// Puts a mandatory 0-20 ms delay between each macro execution (depending on the platform).
 fn keypress_executor_sender(mut rchan_execute: UnboundedReceiver<rdev::EventType>) {
     loop {
-        plugin::util::send(&rchan_execute.blocking_recv().unwrap());
+        let received_event = match &rchan_execute.blocking_recv() {
+            Some(event) => *event,
+            None => {
+                error!("Failed to receive an event!");
+                thread::sleep(time::Duration::from_millis(200));
+                continue;
+            }
+        };
+        plugin::util::direct_send_event(&received_event)
+            .unwrap_or_else(|err| error!("Error directly sending an event to keyboard: {}", err));
 
-        //Windows requires a delay between each macro execution.
-        #[cfg(any(target_os = "macos", target_os = "linux"))]
+        //MacOS and Linux require a delay between each macro execution.
+        #[cfg(not(target_os = "windows"))]
         thread::sleep(time::Duration::from_millis(10));
-
-        //MacOS requires some strange delays so putting it here just in case.
-        #[cfg(target_os = "windows")]
-        thread::sleep(time::Duration::from_millis(1));
```

Comment:
No need for a manual delay on Windows it seems, tested and works well. MacOS/Linux still need it

Reply:
Tweaked later on, addressed